lib/server/controllers/logs.controller.js

```diff
@@ -281,6 +283,40 @@ const getLogById = async (request, response, next) => {
     }
 };
 
+/**
+ * Endpoint to list all the root logs
+ *
+ * @param {Object} request the express request
+ * @param {Object} response the express response
+ * @return {void}
+ */
+const getRootLogs = async (request, response) => {
+    const value = await dtoValidator(GetAllLogsDto, request, response);
+
+    if (value) {
+        try {
+            const { query: { page: { limit = ApiConfig.pagination.limit } = {} } } = value;
+
+            // Get the logs for the current page
+            const { count, logs } = await logService.getAllRootLogs({ ...value, query: { ...value.query, page: { limit } } });
```

Comment:
the services should have no notion of query, it's business layer only (query is an API notion).
Here, what you need to pass to service is directly the offset and limit.

Reply:
done